### PR TITLE
[New] Substack loader

### PR DESCRIPTION
**Pull request by sidmohanty11**

### CODE REVIEW

File: embedchain/loaders/substack.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import concurrent.futures
+import hashlib
+import logging
+
+import requests
+
+try:
```
Comment (deven298) [inline embedchain/loaders/substack.py:7]: Move this inside the class.

File: embedchain/loaders/substack.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+        doc_id = hashlib.sha256((" ".join(links) + url).encode()).hexdigest()
+
+        def load_link(link):
+            try:
+                each_load_data = requests.get(link)
+                each_load_data.raise_for_status()
+
+                data = self.serialize_response(each_load_data)
+                data = str(data).encode()
```
Comment (deven298) [inline embedchain/loaders/substack.py:46]: Why `.encode()`?

File: embedchain/loaders/substack.py (hunk 1 above)
Comment (sidmohanty11) [reply embedchain/loaders/substack.py:6]: Thanks!

File: embedchain/loaders/substack.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import logging
+
+import requests
+
+try:
+    from bs4 import BeautifulSoup
+    from bs4.builder import ParserRejectedMarkup
+except ImportError:
+    raise ImportError(
+        'Sitemap requires extra dependencies. Install with `pip install --upgrade "embedchain[dataloaders]"`'
```
[the remark this answers is not in the document]
Comment (sidmohanty11) [reply embedchain/loaders/substack.py]: Thanks! Missed it

File: embedchain/loaders/substack.py (hunk 1 above)
Comment (deshraj) [reply embedchain/loaders/substack.py:5]: This still seems to be unresolved?

File: docs/data-sources/substack.mdx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+To add any Substack data sources to your app, just add the sitemap.xml of that url as the source and set the data_type to `substack`.
+
+```python
+from embedchain import Pipeline as App
+
+app = App()
+
+# source: for any substack just add the sitemap.xml url
+app.add('https://www.lennysnewsletter.com/sitemap.xml', data_type='substack')
```
Comment (deshraj) [inline docs/data-sources/substack.mdx:13]: Do the user have to enter the full url with `sitemap.xml` ? I think we should just assume that sitemap.xml is present and then try to fetch it. If we fail, then we should return error to the user

Comment (deshraj) [reply docs/data-sources/substack.mdx:13]: I am going to merge it for now but lets do this change in a follow up PR.

**Review by deshraj — APPROVED** [review]

**Pull request merged**